src/main/webapp/app/shared/components/checkbox/checkbox.component.html

```diff
@@ -0,0 +1 @@
+<mat-checkbox (click)="pushChanges('')">{{name}}</mat-checkbox>
```

Comment:
Why did you decided to create a checkbox component when its only wrapping the mat-checkbox? :)

Also the click handler seems odd as it does not push any changes but an empty string.

Reply:
Great catch regarding the click handler. We don't need this at this point. 

Regarding the mat-checkbox I had some plans to make this a global component by adding few more options, then this can be used by other areas also. Using shared components will give us the freedom to update to any libraries, for example, if we use PrimeNG later we have to change in one place.